test/recorder.spec.js

```diff
@@ -61,16 +56,20 @@ describe.skip(USES_HOOKS)('Recorder', function() {
     state.context = await state.browser.newContext();
     state.output = new WritableBuffer();
     const debugController = state.context._initDebugModeForTest({ recorderOutput: state.output });
+    state.page = await state.context.newPage();
+    state.setContent = async (content) => {
+      await state.page.setContent(content);
+      await debugController._ensureInstalledInFrame(state.page.mainFrame());
```

Comment:
Isn't it supposed to happen automatically in non-test environment as well? I'd expect something like `_initDebugModeForTest()` to be sufficient here, is the problem that the script is not properly reinstalled after changing the content?

Reply:
IIUC, `document.open()` removes any document listeners. That means, recorder may not work in frames that did `document.open()` after we installed the recorder. Since we lack the precise notification about `document.open` behavior, I can probably separately fix that by reinstalling the listeners on a timeout, but the test will have to be fixed separately.